### PR TITLE
homedir fixes

### PR DESCRIPTION
**Pull request by minrk**
- don't relocate home if home exists (e.g. mounted volume)
- respect workdir, only cd to new home if we were in /home/jovyan/

#455 broke SystemUserSpawner, where workdir is already set and /home/$NB_USER is a mounted volume that should not be modified.

### CODE REVIEW

**Comment by parente** [comment]
Whoopsie. That was a merge misclick in #455.

File: base-notebook/start.sh [hunk 1]
```diff
@@ -6,13 +6,26 @@ set -e
...
-    # Switch to the root of the container in case we start adjusting paths
-    # that impact the default working directory
-    cd /
 
     # Handle username change. Since this is cheap, do this unconditionally
     echo "Set username to: $NB_USER"
-    usermod -m -d /home/$NB_USER -l $NB_USER jovyan
+    usermod -d /home/$NB_USER -l $NB_USER jovyan
+
+    # handle home and workding directory if the username changed
```
Comment (parente) [inline base-notebook/start.sh]: working

**Comment by parente** [comment]
The number of options we support and the scenarios we need to keep in mind when making changes no longer fits in my head between contributions here. I'll spend some time on #406 to alleviate that pain.

Thanks for the save @minrk

File: base-notebook/start.sh (hunk 1 above)
Comment (minrk) [reply base-notebook/start.sh]: Thanks! fixed.

**Comment by parente** [comment]
https://cloud.docker.com/app/jupyter/repository/docker/jupyter/all-spark-notebook/timeline failed to build (out of disk). I've restarted it.

**Pull request merged**